### ISSUE TITLE
Uzebox sound engine support

### ISSUE THREAD

**Issue by danboid** [bintracker/bintracker#21]
The Uzebox open source games console still lacks an easy way to compose music for it. There is, however, Uzebox patch studio which is an open source implementation of the Uzebox sound engine but it lacks MIDI input or any kind of tracker or sequencer

https://github.com/nebososo/uzebox-patch-studio

Might you consider adding Uzebox support to bintracker?

Thanks

**Comment by danboid** [comment]
I have successfully built bintracker under Debian trixie.

I've not seen a Linux app use Tk since the 90s!

**Comment by utz82** [comment]
Yes, that sounds like an interesting target. The main blocking issue that I see is that Uzebox emulation in MAME is currently marked as "preliminary", which means it's basically not ready for use. Do you know anybody you could convince to work on that?

Another option could be to either use the CUzebox emulator, Uzebox Patch Studio, or an actual Uzebox as a backend, but then those would need to be scripted to speak to Bintracker somehow. Also, Bintracker would need to support backends other than MAME, but that's something I want to work on soon anyway.

Also, Bintracker still lacks the graphical editing capabilities. While not strictly required, working with Uzebox's wavetable-centric sound wouldn't be much fun without.

So, in short, yes, I'm open to supporting Uzebox in Bintracker, but I see a few roadblocks that need to be moved out of the way first.

**Comment by danboid** [comment]
Excellent!

I'm only a beginner C dev so adding whatever functionality you require to cuzebox is probably beyond my abilities but I shall post your response to the Uzebox forums and get back to you when the more experienced Uzebox developers have given their input.

I'm sure they'll be happy to add whatever features you might need to cuzebox as they've been discussing the need for a Uzebox tracker for over a decade now.
